hs_file_types/utils.py

```diff
@@ -226,20 +231,159 @@ def get_logical_file_type(res, file_id, hs_file_type=None, fail_feedback=True):
     return logical_file_type_class
 
 
+def get_logical_file(agg_type_name):
+    file_type_map = {"GeographicRasterAggregation": GeoRasterLogicalFile,
+                     "SingleFileAggregation": GenericLogicalFile,
+                     "FileSetAggregation": FileSetLogicalFile,
+                     "MultidimensionalAggregation": NetCDFLogicalFile,
+                     "GeographicFeatureAggregation": GeoFeatureLogicalFile,
+                     "ReferencedTimeSeriesAggregation": RefTimeseriesLogicalFile,
+                     "TimeSeriesAggregation": TimeSeriesLogicalFile,
+                     }
+    return file_type_map[agg_type_name]
+
+
+def is_aggregation_metadata_file(file):
+    return file.name.endswith('_meta.xml')
+
+
+def is_map_file(file):
+    return file.name.endswith('_resmap.xml') or file.name == 'resourcemap.xml'
+
+
+def is_resource_metadata_file(file):
+    return os.path.basename(file.name) == 'resourcemetadata.xml'
+
+
+def identify_and_ingest_metadata_files(resource, files):
+    from hs_file_types.utils import identify_metadata_files
+    res_files, meta_files = identify_metadata_files(files)
+    from hs_file_types.utils import ingest_metadata_files
+    ingest_metadata_files(resource, meta_files)
+
+
+def ingest_metadata_files(resource, meta_files):
+    # refresh to pick up any new possible aggregations
+    resource_metadata_file = None
+    for f in meta_files:
+        if is_resource_metadata_file(f):
+            resource_metadata_file = f
+        elif is_aggregation_metadata_file(f):
+            ingest_logical_file_metadata(f, resource)
+    if resource_metadata_file:
+        resource.refresh_from_db()
+        graph = Graph().parse(data=resource_metadata_file.read())
+        try:
+            with transaction.atomic():
+                resource.metadata.delete_all_elements()
+                resource.metadata.ingest_metadata(graph)
+        except:
+            # logger.exception("Error processing resource metadata file")
+            raise
```

Comment:
resource_metadata_file variable is not needed. Lines 274 to 282 can go directly to where line 270 is, which would make the logic a bit clearer and simpler. Also, line 281 can be deleted.

Reply:
I'm intentionally processing the resource metadata file last.  This is because aggregations can overwrite resource metadata.  I'll add a comment to make this clear.